### PR TITLE
[DOC] Update plot_advanced_decoding_scikit.py

### PR DESCRIPTION
**Pull request by emdupre**
- Partially addresses #6265

Use of AI tools:
- [ ] LLM tools were used to generate at least part of the content of this pull-request.
- [X] No LLM tools were used to generate any part of the content of this pull-request.

Changes proposed in this pull request:

- Updates the _Advanced decoding using scikit learn_ example
- Provides better linking to scikit-learn docs, improvements for readability

### CODE REVIEW

**Comment by emdupre** [comment]
~Sorry, I realized I might have been better off splitting these up !
I'll go ahead and open the next couple of examples in new PRs, but let me know if you want me to also break out these two updates into separate PRs.~
EDIT : I went ahead and split these up to ease reviewing ! 🙏

**Review by bthirion — COMMENTED** [review]
The PR LGTM but there seems to be a doc build failure.

**Comment by emdupre** [comment]
> The PR LGTM but there seems to be a doc build failure.

For some reason, `sklearn.model_selection.LeaveOneGroupOut` wasn't resolving. I changed the text to move the anchor to the [User Guide entry on cross-validation](https://scikit-learn.org/stable/modules/cross_validation.html), which resolves cleanly.

**Comment by emdupre** [comment]
👉  [The built example is visible here](https://output.circle-artifacts.com/output/job/85d6adcf-ea76-4f9d-88e0-076937ca3322/artifacts/0/dev/auto_examples/07_advanced/plot_advanced_decoding_scikit.html)

**Review by bthirion — APPROVED** [review]
LGTM, thx !

File: examples/07_advanced/plot_advanced_decoding_scikit.py [hunk 1]
```diff
@@ -1,32 +1,38 @@
...
-# .......................
-#
-
+# Download the data
+# .................
 # The :func:`~nilearn.datasets.fetch_haxby` function will download the
-# Haxby dataset composed of fMRI images in a Niimg,
-# a spatial mask and a text document with label of each image
+# Haxby dataset object, whose attributes include
+# the fMRI images as Niimg objects (``func``)
```
Comment (Remi-Gau) [inline examples/07_advanced/plot_advanced_decoding_scikit.py]:
```suggestion
# the fMRI images as Niimg objects (``func``),
```

File: examples/07_advanced/plot_advanced_decoding_scikit.py
Comment (Remi-Gau) [inline examples/07_advanced/plot_advanced_decoding_scikit.py:92]:
```suggestion
    standardize="zscore_sample",
    memory="nilearn_cache",
```

There has been a massive drop in performance in this example between 0.14.0 and 0.13.1

https://nilearn.github.io/0.13.1/auto_examples/07_advanced/plot_advanced_decoding_scikit.html

I think that here too this is due to me remove 
standardize="zscore_sample" from places where it should have stayed.

Let's readd it and see it, if it gets back to normal.

Comment (emdupre) [reply examples/07_advanced/plot_advanced_decoding_scikit.py:92]: This does improve performance (tuned SVC accuracy is now up to 0.799 and the weight maps are non-empty) !
I'm also setting `draw_cross=False` since it obscures some of the weights.

**Comment by emdupre** [comment]
👉 [The built example is visible here](https://output.circle-artifacts.com/output/job/66a56a01-0bbe-4f1e-baa0-96eebb449442/artifacts/0/dev/auto_examples/07_advanced/plot_advanced_decoding_scikit.html) and ready for re-review !

File: examples/07_advanced/plot_advanced_decoding_scikit.py
Comment (Remi-Gau) [reply examples/07_advanced/plot_advanced_decoding_scikit.py:92]: OK let's not tweak further for now but I think we had even better performance before so we should figure what happened.

Comment (emdupre) [reply examples/07_advanced/plot_advanced_decoding_scikit.py:92]: Should we open an issue to track ?

**Review by Remi-Gau — APPROVED** [review]
good to go
I will try to investigate further why we have a drop in performance

File: examples/07_advanced/plot_advanced_decoding_scikit.py
Comment (Remi-Gau) [reply examples/07_advanced/plot_advanced_decoding_scikit.py:92]: will do

**Comment by Remi-Gau** [comment]
wonder if change of performance is not also related to the change of cv from cv=5 to leave one group out

**Comment by bthirion** [comment]
> wonder if change of performance is not also related to the change of cv from cv=5 to leave one group out

Yes, quite likely
